### DISCUSSION TITLE
Hydrogen Package not Rebuilding (Possibly after Windows Update)

### DISCUSSION THREAD

**Discussion by eugenetkj98** [atom/atom#22194] [Q&A]
Has anybody ever encountered an issue where atom fails to rebuild the Hydrogen package after a windows update? It was working fine this morning until after updating my system. I've tried to reinstall VSCode and Visual Studio and the likes, but it doesn't seem to be working. Any ideas? I've included the error log below:

`> zeromq@5.2.0 install C:\Users\eugen\AppData\Local\Temp\apm-install-dir-2021314-12992-1nwzzx7.y3lo\node_modules\Hydrogen\node_modules\zeromq
> node scripts/prebuild-install.js || (node scripts/preinstall.js && node-gyp rebuild)

prebuild-install WARN install No prebuilt binaries found (target=9.4.4 runtime=electron arch=x64 libc= platform=win32)

Downloading libzmq for Windows
Download finished

C:\Users\eugen\AppData\Local\Temp\apm-install-dir-2021314-12992-1nwzzx7.y3lo\node_modules\Hydrogen\node_modules\zeromq>if not defined npm_config_node_gyp (node "C:\Users\eugen\AppData\Local\atom\app-1.56.0\resources\app\apm\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Users\eugen\AppData\Local\atom\app-1.56.0\resources\app\apm\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild ) 
Building the projects in this solution one at a time. To enable parallel build, please add the "-m" switch.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.targets(439,5): error MSB8020: The build tools for Visual Studio 2015 (Platform Toolset = 'v140') cannot be found. To build using the v140 build tools, please install Visual Studio 2015 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\Users\eugen\AppData\Local\Temp\apm-install-dir-2021314-12992-1nwzzx7.y3lo\node_modules\Hydrogen\node_modules\zeromq\build\zmq.vcxproj]

npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated json-parser@1.1.5: `json-parser` is deprecated. Please use `comment-json` instead
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated highlight.js@9.15.10: Version no longer supported. Upgrade to @latest
gyp ERR! build error 
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Users\eugen\AppData\Local\atom\app-1.56.0\resources\app\apm\node_modules\npm\node_modules\node-gyp\lib\build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:200:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Windows_NT 10.0.19042
gyp ERR! command "C:\\Users\\eugen\\AppData\\Local\\atom\\app-1.56.0\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\eugen\\AppData\\Local\\atom\\app-1.56.0\\resources\\app\\apm\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\eugen\AppData\Local\Temp\apm-install-dir-2021314-12992-1nwzzx7.y3lo\node_modules\Hydrogen\node_modules\zeromq
gyp ERR! node -v v12.4.0
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
npm WARN enoent ENOENT: no such file or directory, open 'C:\Users\eugen\AppData\Local\Temp\apm-install-dir-2021314-12992-1nwzzx7.y3lo\package.json'
npm WARN apm-install-dir-2021314-12992-1nwzzx7.y3lo No description
npm WARN apm-install-dir-2021314-12992-1nwzzx7.y3lo No repository field.
npm WARN apm-install-dir-2021314-12992-1nwzzx7.y3lo No README data
npm WARN apm-install-dir-2021314-12992-1nwzzx7.y3lo No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zeromq@5.2.0 install: `node scripts/prebuild-install.js || (node scripts/preinstall.js && node-gyp rebuild)`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zeromq@5.2.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\eugen\.atom\.apm\_logs\2021-04-14T06_35_29_644Z-debug.log`

**Comment by icecream17** [comment]
#22191 seems vaguely similar, did you try uninstalling/reinstalling hydrogen?

**Reply by eugenetkj98** [reply]
That's what I did try, but at the current point, Hydrogen doesn't appear to be installed at all... so nothing to uninstall on that end

**Comment by eugenetkj98** [comment]
I'm actually on windows though, so I don't think my computer uses Xcode?


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by ptoche** [comment]
An update has just been released and it's working on my system. You should try again now.
